packages/core/src/notifications/panelNode.ts

```diff
@@ -126,15 +130,125 @@ export class NotificationsNode implements TreeNode {
+
+            if (await fs.existsFile(tempFilePath)) {
+                // If file exist, make sure it has write permission (0o644)
+                await fs.chmod(tempFilePath, 0o644)
+            }
+
+            await fs.writeFile(tempFilePath, content)
+
+            // Set the file permissions to read-only (0o444)
+            await fs.chmod(tempFilePath, 0o444)
+
+            // Now, open the document
+            const document = await vscode.workspace.openTextDocument(tempFilePath)
+
+            const options: vscode.TextDocumentShowOptions = {
+                viewColumn: vscode.ViewColumn.Beside,
+                preserveFocus: true,
+                preview: true,
+            }
+
+            await vscode.window.showTextDocument(document, options)
+        } catch (error) {
+            throw new ToolkitError(`Error showing text document: ${error}`)
+        }
+    }
+
+    /**
+     * Renders information window with the notification's content and buttons.
+     * Can be either a blocking modal or a bottom-right corner toast
+     * Handles the button click actions based on the button type.
+     */
+    public async showInformationWindow(notification: ToolkitNotification, type: string = 'toast') {
+        const isModal = type === 'modal'
+
+        // modal has to have defined actions(buttons)
+        const buttons = notification.uiRenderInstructions.actions ?? []
+        const buttonLabels = buttons.map((actions) => actions.displayText['en-US'])
+        const detail = notification.uiRenderInstructions.content['en-US'].description
+
+        // we use toastPreview to display as titlefor toast, since detail won't be shown
+        const title = isModal
+            ? notification.uiRenderInstructions.content['en-US'].title
+            : (notification.uiRenderInstructions.content['en-US'].toastPreview ??
+              notification.uiRenderInstructions.content['en-US'].title)
+
+        const selectedText = await vscode.window.showInformationMessage(
+            title,
+            { modal: isModal, detail },
+            ...buttonLabels
         )
+
+        if (selectedText) {
+            const selectedButton = buttons.find((actions) => actions.displayText['en-US'] === selectedText)
+            // Different button options
+            if (selectedButton) {
+                switch (selectedButton.type) {
+                    case 'openTxt':
+                        await this.showReadonlyTextDocument(
+                            notification.uiRenderInstructions.content['en-US'].description
+                        )
+                        break
+                    case 'updateAndReload':
+                        await this.updateAndReload(notification.displayIf.extensionId)
+                        break
+                    case 'openUrl':
+                        if (selectedButton.url) {
+                            await vscode.env.openExternal(vscode.Uri.parse(selectedButton.url))
```

Comment:
use our `openUrl` wrapper unless the URL contains PII https://github.com/aws/aws-toolkit-vscode/blob/cee0fa3c1ab320f00a4b43fa3e57a44efe3370e3/packages/core/src/shared/utilities/vsCodeUtils.ts#L217